machine-learning/tutorials/TransferLearningTF/Program.cs

```diff
@@ -16,150 +13,106 @@ class Program
             Console.WriteLine("=============== Training classification model ===============");
-            // Create and train the model based on the dataset that has been loaded, transformed.
+            // Create and train the model
             // <SnippetTrainModel>
-            ITransformer model = estimator.Fit(data);
+            ITransformer model = pipeline.Fit(trainingData);
             // </SnippetTrainModel>
 
-            // Process the training data through the model
-            // This is an optional step, but it's useful for debugging issues
-            // <SnippetTransformData>
-            var predictions = model.Transform(data);
-            // </SnippetTransformData>
-
-            // Create enumerables for both the ImageData and ImagePrediction DataViews 
-            // for displaying results
-            // <SnippetEnumerateDataViews>
-            var imageData = mlContext.Data.CreateEnumerable<ImageData>(data, false, true);
-            var imagePredictionData = mlContext.Data.CreateEnumerable<ImagePrediction>(predictions, false, true);
-            // </SnippetEnumerateDataViews>
+            // Generate predictions from the test data, to be evaluated
+            // <SnippetLoadAndTransformTestData>
+            IDataView testData = mlContext.Data.LoadFromTextFile<ImageData>(path: _testTagsTsv, hasHeader: false);
+            IDataView predictions = model.Transform(trainingData);
 
-            // <SnippetCallDisplayResults1>
+            // Create an IEnumerable for the predictions for displaying results
+            IEnumerable<ImagePrediction> imagePredictionData = mlContext.Data.CreateEnumerable<ImagePrediction>(predictions, false);
             DisplayResults(imagePredictionData);
-            // </SnippetCallDisplayResults1>
+            // </SnippetLoadAndTransformTestData>
 
-            // Get some performance metrics on the model using training data
+            // Get performance metrics on the model using training data
             Console.WriteLine("=============== Classification metrics ===============");
 
             // <SnippetEvaluate>           
-            var multiclassContext = mlContext.MulticlassClassification;
-            var metrics = multiclassContext.Evaluate(predictions, labelColumnName: LabelTokey, predictedLabelColumnName: "PredictedLabel");
+            MulticlassClassificationMetrics metrics =
+                mlContext.MulticlassClassification.Evaluate(predictions,
+                  labelColumnName: "LabelKey",
+                  predictedLabelColumnName: "PredictedLabel");
             // </SnippetEvaluate>
 
             //<SnippetDisplayMetrics>
             Console.WriteLine($"LogLoss is: {metrics.LogLoss}");
             Console.WriteLine($"PerClassLogLoss is: {String.Join(" , ", metrics.PerClassLogLoss.Select(c => c.ToString()))}");
             //</SnippetDisplayMetrics>
 
+            mlContext.Model.Save(model, trainingData.Schema, "model.zip");
```

Comment:
Not sure if this step is going to be part of the tutorial since it does not have a snippet tag.

Reply:
Good catch. I added this line for debugging. Will remove it